core/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -360,22 +359,21 @@ public void finalizeRecovery() {
         cancellableThreads.checkForCancel();
         StopWatch stopWatch = new StopWatch().start();
         logger.trace("[{}][{}] finalizing recovery to {}", indexName, shardId, request.targetNode());
-
-
         cancellableThreads.execute(recoveryTarget::finalizeRecovery);
 
         if (isPrimaryRelocation()) {
+            cancellableThreads.checkForCancel();
+            logger.trace("[{}][{}] performing relocation hand-off to {}", indexName, shardId, request.targetNode());
+            try {
+                cancellableThreads.execute(() -> shard.relocated("to " + request.targetNode()));
+            } catch (Throwable e) {
+                logger.debug("[{}][{}] completing relocation hand-off to {} failed", indexName, shardId, request.targetNode(), e);
```

Comment:
Good catch. I was surprised to see that this was not caught by the logger usage checker. After digging I noticed that the checker was accidentally disabled since it was added. I've opened #19126 to address this.